/*********************************************************************************
 * Modifications Copyright 2017-2019 eBay Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *    https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 *********************************************************************************/

syntax = "proto3";

import "raft_types.proto";

package nuraft_mesg;

message RaftGroupMsg {
    string                  group_id            = 1;
    RaftMessage             msg                 = 2;
    string                  intended_addr       = 3;
    string                  group_type          = 4;
}
service Messaging { rpc RaftStep(RaftGroupMsg) returns (RaftGroupMsg); }
